syntax = "proto3";

package scene;

service SceneService {
    rpc UpdateScene (SceneConfig) returns (SceneResponse);
    rpc GetScene (Empty) returns (SceneConfig);
    rpc ClearScene (Empty) returns (SceneResponse);
}

message Empty {}

message Point {
    double x = 1;
    double y = 2;
}

message Rectangle {
    Point topleft = 1;
    Point bottomright = 2;
}

message SceneConfig {
    Point start = 1;
    Point end = 2;
    repeated Rectangle obstacles = 3;
    repeated Rectangle safezones = 4;
}

message SceneResponse {
    bool success = 1;
    string message = 2;
}
